File: 2730. Find the Longest Semi-Repetitive Substring/Solution.py
class Solution:
    def longestSemiRepetitiveSubstring(self, s: str) -> int:
        left, doubleRight, high, found = 0, 0, 1, False
        for right in range(1, len(s)):
            if s[right] == s[right - 1]:
                high = max(high, right - left) if found else max(high, right - left + 1)
                left = doubleRight
                doubleRight = right
                found = True
            else:
                high = max(high, right - left + 1)
        return high

File: 2730. Find the Longest Semi-Repetitive Substring/Solution.cpp
class Solution {
public:
    int longestSemiRepetitiveSubstring(string s) {
        int left = 0, doubleRight = 0, high = 1;
        bool found = false;
        for (int right = 1; right < s.length(); right++) {
            if (s[right] == s[right - 1]) {
                high = (found) ? max(high, right - left) : max(high, right - left + 1);
                left = doubleRight;
                doubleRight = right;
                found = true;
            } else {
                high = max(high, right - left + 1);
            }
        }
        return high;
    }
};
